broker/ga4gh/broker/endpoints/repositories.py

```diff
@@ -0,0 +1,287 @@
+        generating unique identifier.)
+        - Generates access_token of 32 characters. (access_token will be
+        used later for operations like creating a new build, updating build
+        completion.)
+        - Inserts object in mongodb.
+        - Remove url from the repository_object
+        - Returns id and access_token for the repository_object created.
+
+    Raises:
+        URLNotFound: Raised when request data does not have url string.
+        InternalServerError: Raised when it's unable to create unique
+        identifier for the repository_object.
+
+
+    """
+    retries = 3
+    repository_object = {}
+    try:
+        repository_object['url'] = data['url']
+        # Needs to verify validity of repository url.
+    except KeyError:
+        raise URLNotFound
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['brokerStore'].
+        collections['repositories'].client
+    )
+    id_length = (
+        current_app.config['FOCA'].endpoints['repository']['id_length']
+    )
+    id_charset: str = (
+        current_app.config['FOCA'].endpoints['repository']['id_charset']
+    )
+    try:
+        id_charset = eval(id_charset)
+    except Exception:
+        id_charset = ''.join(sorted(set(id_charset)))
+
+    for i in range(retries + 1):
+        logger.debug(f"Trying to insert/update object: try {i}")
+
+        repo_id = generate_id(
+            charset=id_charset,
+            length=id_length,
+        )
+        access_token = str(generate_id(id_charset, 32))
+        repository_object['id'] = repo_id
+        repository_object['access_token'] = access_token
+        try:
+            db_collection.insert_one(repository_object)
+            break
+        except DuplicateKeyError:
+            logger.error(f"DuplicateKeyError ({repo_id}): Key generated"
+                         f" is already present.")
+            continue
+    else:
+        logger.error(
+                    f"Could not generate unique identifier."
+                    f" Tried {retries + 1} times."
+                )
+        raise InternalServerError
+    if repository_object is not None and '_id' in repository_object:
+        del repository_object['_id']
+        del repository_object['url']
+    logger.info(f"Added object with '{repository_object}'.")
+    return repository_object
+
+
+def get_repositories():
+    """Get available Repositories.
+
+    Returns:
+        list of repository_objects
+
+    Description:
+        - Fetches all repositories from the mongodb.
+        - Removes subscription_list if present.
+        - Returns list of repository_objects.
+
+    Raises:
+        RepositoryNotFound: Raised when there is no registered repository in
+        the broker.
+
+    """
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['brokerStore'].
+        collections['repositories'].client
+    )
+
+    cursor = db_collection.find(
```

Comment:
Would be good to create an issue for paginating these responses as there will be many repos. Same goes for builds and the other resources. No need to do this now, of course, but good to have the issue so we won't forget